cirq/ops/linear_combinations.py

```diff
@@ -487,17 +487,50 @@ def __neg__(self):
         factory = type(self)
         return factory(-self._linear_dict)
 
-    def __imul__(self, a: value.Scalar):
-        self._linear_dict *= a
-        return self
-
-    def __mul__(self, a: value.Scalar):
-        result = self.copy()
-        result *= a
-        return result
+    def __imul__(self, other: PauliSumLike):
+        if isinstance(other, (float, int, complex)):
+            self._linear_dict *= other
+            return self
+        # No convenient way to mutate paulistring members.
+        return self.__mul__(other)
```

Comment:
We need to be consistent about whether or not `*=` modifies the origin or makes a copy. This is a very surprising change in aliasing behavior based on the type of the input. The method has to stick with the current behavior of modifying inline.

Note that you shouldn't be mutating the pauli strings themselves, but you should be changing the keys and values that are in the dictionary.

I think the correct behavior here would be equivalent to `self._linear_dict = self.__mul__(other)._linear_dict`. But you should implement mul in terms of imul instead of vice versa.

Reply:
I think by convention the `*=` *must* modify the original (?). Otherwise you shouldn't support `*=`

Reply:
okay good catch; fixed it.